File: SoSemiVigelant.Provider/Entities/Auction.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace SoSemiVigelant.Provider.Entities
{
    class Auction
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("lot")]
        public string Lot { get; set; }

        [JsonProperty("date_estimated")]
        private long estimated;
        public DateTime Estimated { get
            {
                return new DateTime(estimated);
            }
            set
            {
                estimated = value.Ticks;
            }
        }

        [JsonProperty("image")]
        public string ImageName { get; set; }

        [JsonProperty("start_bid")]
        public int StartBin { get; set; }

        [JsonProperty("current_bid")]
        public int CurrentBid { get; set; }

        [JsonProperty("bid_amount")]
        public int BidAmount { get; set; }

        [JsonProperty("shipping_info_quick")]
        public string ShippingInfoShort { get; set; }

        [JsonProperty("shipping_info")]
        public string ShippingInfo { get; set; }

        [JsonProperty("bin_value")]
        public int BinValue { get; set; }

        [JsonProperty("date_published")]
        private long published;
        public DateTime Published
        {
            get
            {
                return new DateTime(published);
            }
            set
            {
                published = value.Ticks;
            }
        }

        [JsonProperty("thumb_is_generic")]
        public bool ThumbIsGeneric { get; set; }

        [JsonProperty("with_post")]
        public bool WithPost { get; set; }

        [JsonProperty("city")]
        public string City { get; set; }

        [JsonProperty("seller_ip")]
        public string SellerIp { get; set; }

        [JsonProperty("seller")]
        public Seller Seller { get; set; }
    }
}

File: SoSemiVigelant.Auctioneer/src/storage/entities/Auction.ts
import { Schema, Document, Model, model, Types } from 'mongoose';
import { User } from './User';

export interface AuctionModel {
    [index: string]: any;

    id: number;
    name: string;
    estimated: Date;
    image: string;
    startBid: number;
    currentBid: number;
    bidAmount: number;
    shippingInfoShort: string;
    shippingInfo: string;
    binValue: number;
    published: Date;
    thumbIsGeneric: boolean;
    withPost: boolean;
    city: string;
    sellerIp: string;
    seller: number;
    description: string;
    isActive: boolean;
}

export const AuctionSchema: Schema = new Schema({
    id: {
        type: Number,
        required: true,
        unique: true
    },
    name: String,
    estimated: Date,
    image: String,
    startBid: Number,
    currentBid: Number,
    bidAmount: Number,
    shippingInfoShort: String,
    shippingInfo: String,
    binValue: Number,
    published: Date,
    thumbIsGeneric: Boolean,
    withPost: Boolean,
    city: String,
    sellerIp: String,
    seller: { type: Schema.Types.ObjectId, ref: 'users' },
    description: String,
    isActive: { type: Boolean, default: true }
}, {
    toObject: {
        transform: function (doc, ret) {
            delete ret._id;
            delete ret.__v;
        }
    },
    toJSON: {
        transform: function (doc, ret) {
            delete ret._id;
            delete ret.__v;
        }
    }
});

export const Auction: Model<AuctionModel & Document> = model<AuctionModel & Document>('auctions', AuctionSchema);

export const AuctionMap: Record<keyof AuctionModel, string | { model: Model<Document>, path: string, key: string }> = {
    id: 'id',
    name: 'lot',
    estimated: 'date_estimated',
    image: 'image',
    startBid: 'start_bid',
    currentBid: 'current_bid',
    bidAmount: 'bid_amount',
    shippingInfoShort: 'shipping_info_quick',
    shippingInfo: 'shipping_info',
    binValue: 'bin_value',
    published: 'date_published',
    thumbIsGeneric: 'thumb_is_generic',
    withPost: 'with_post',
    city: 'city',
    sellerIp: 'seller_ip',
    seller: { model: User, path: 'id', key: 'seller' }
};
